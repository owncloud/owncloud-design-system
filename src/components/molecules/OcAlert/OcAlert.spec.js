import { shallowMount, mount } from "@vue/test-utils"
import Alert from "./OcAlert.vue"

const testSlots = { default: '<p class="slot-message">Test message</p>' }

describe("OcAlert", () => {
  it("displays correct message", () => {
    const wrapper = shallowMount(Alert, {
      slots: testSlots,
    })

    expect(wrapper.find(".slot-message").text()).toMatch("Test message")
    expect(wrapper).toMatchSnapshot()
  })

  it("hides the close button if disabled", () => {
    const wrapper = shallowMount(Alert, {
      propsData: {
        noClose: true,
      },
      slots: testSlots,
    })

    expect(wrapper.props().noClose).toBe(true)
    expect(wrapper.findAll(".oc-alert-close").length).toBe(0)
    expect(wrapper).toMatchSnapshot()
  })

  it("sets correct variation", () => {
    const wrapper = shallowMount(Alert, {
      propsData: {
        variation: "primary",
      },
      slots: testSlots,
    })

    expect(wrapper.props().variation).toMatch("primary")
    expect(wrapper.classes()).toContain("oc-alert-primary")
    expect(wrapper).toMatchSnapshot()
  })

  it("emits a close event", () => {
    const wrapper = mount(Alert, {
      slots: testSlots,
    })

    wrapper.find(".oc-alert-close").trigger("click")
    expect(wrapper.emitted().close).toBeTruthy()
  })
})
