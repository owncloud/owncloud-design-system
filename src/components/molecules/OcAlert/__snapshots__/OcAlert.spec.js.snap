// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OcAlert displays correct message 1`] = `
<div uk-alert="" class="oc-alert oc-alert-passive">
  <oc-button-stub type="button" size="medium" arialabel="Close alert" variation="passive" appearance="raw" justifycontent="center" gapsize="medium" class="uk-alert-close">
    <oc-icon-stub name="close" filltype="fill" accessiblelabel="" type="span" size="small" variation="inverse"></oc-icon-stub>
  </oc-button-stub>
  <p class="slot-message">Test message</p>
</div>
`;

exports[`OcAlert hides the close button if disabled 1`] = `
<div uk-alert="" class="oc-alert oc-alert-passive">
  <!---->
  <p class="slot-message">Test message</p>
</div>
`;

exports[`OcAlert sets correct variation 1`] = `
<div uk-alert="" class="oc-alert oc-alert-primary">
  <oc-button-stub type="button" size="medium" arialabel="Close alert" variation="passive" appearance="raw" justifycontent="center" gapsize="medium" class="uk-alert-close">
    <oc-icon-stub name="close" filltype="fill" accessiblelabel="" type="span" size="small" variation="inverse"></oc-icon-stub>
  </oc-button-stub>
  <p class="slot-message">Test message</p>
</div>
`;
